platform/orion/core/generic/committer/committer.go

```diff
@@ -132,9 +133,37 @@ func (c *committer) Commit(block *types.AugmentedBlockHeader) error {
 			}
 			event.Committed = true
 		default:
-			// rollback
-			if err := c.vault.DiscardTx(txID); err != nil {
-				return errors.WithMessagef(err, "failed to discard tx %s", txID)
+			blockNum := block.Header.BaseHeader.Number
+			if logger.IsEnabledFor(zapcore.DebugLevel) {
+				logger.Debugf("transaction [%s] in block [%d] is not valid for fabric [%s], discard!", txID, blockNum, validationCode)
+			}
+
+			vc, err := c.vault.Status(txID)
+			if err != nil {
+				logger.Panicf("failed getting tx's status [%s], with err [%s]", txID, err)
+			}
+			switch vc {
```

Comment:
can we please write this in a way that we dont have a switch inside a switch?

Reply:
okay